relays/ethereum/src/main.rs

```diff
@@ -210,50 +213,55 @@ fn substrate_signing_params(matches: &clap::ArgMatches) -> Result<SubstrateSigni
 }
 
 fn ethereum_sync_params(matches: &clap::ArgMatches) -> Result<EthereumSyncParams, String> {
-	let mut eth_sync_params = EthereumSyncParams::default();
-	eth_sync_params.eth = ethereum_connection_params(matches)?;
-	eth_sync_params.sub = substrate_connection_params(matches)?;
-	eth_sync_params.sub_sign = substrate_signing_params(matches)?;
-	eth_sync_params.metrics_params = metrics_params(matches)?;
+	let instance = instance_params(matches)?;
+	let mut params = EthereumSyncParams::with_instance(instance);
```

Comment:
Why not create the `EthereumSyncParams` object directly here and get rid of the constructor completely? You wouldn't need defaults at all and it would be easier to ensure that all parameters are set to non-default values.

Reply:
The problem lies with `sync_params`. For the two different sync params (Ethereum and Substrate) we require a different "default" for `HeaderSyncParams`. To build this default it requires us to pull in ~5 different constants from the corresponding module. It seemed nicer to keep that as part of the corresponding modules, but I can look into changing it